livekit-agents/livekit/agents/voice_assistant/voice_assistant.py

```diff
@@ -17,11 +18,20 @@
 from .plotter import AssistantPlotter
 from .speech_handle import SpeechHandle
 
-WillSynthesizeAssistantReply = Callable[
+
+PreLLMGenerationCallback = Callable[
     ["VoiceAssistant", ChatContext],
     Union[Optional[LLMStream], Awaitable[Optional[LLMStream]]],
 ]
 
+WillSynthesizeAssistantReply = PreLLMGenerationCallback
+
+PreTTSGenerationCallback = Callable[
```

Comment:
should this be something like?

```
Union[
  Callable[["VoiceAssistant", str], str],
  Callable[["VoiceAssistant", AsyncIterable[str]], AsyncIterable[str]],
]
```

Reply:
It is different, we're OK receiving AsyncIterable[str] and finally returning a str for e.g